### PR TITLE
Add CanGc argument to reflect_dom_object 

### PR DESCRIPTION
**Pull request by willypuzzle**
- Modified fn signature
- modified all fn calls

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #[33684](https://github.com/servo/servo/issues/33684)

- [x] These changes do not require tests because no behaviour was introduced or changed

### CODE REVIEW

**Review by simonwuelker — APPROVED** [review]
Thanks!

File: components/script/dom/webxr/xrviewerpose.rs [hunk 1]
```diff
@@ -175,7 +175,11 @@ impl XRViewerPose {
         let transform: RigidTransform3D<f32, Viewer, BaseSpace> =
             viewer_pose.transform.then(&to_base);
         let transform = XRRigidTransform::new(global, cast_transform(transform), can_gc);
-        let pose = reflect_dom_object(Box::new(XRViewerPose::new_inherited(&transform)), global);
+        let pose = reflect_dom_object(
+            Box::new(XRViewerPose::new_inherited(&transform)),
+            global,
+            CanGc::note(),
```
Comment (simonwuelker) [inline components/script/dom/webxr/xrviewerpose.rs:181]: there is already a `can_gc` argument for this function, so no need to call `CanGc::note`

File: components/script/dom/webxr/xrjointpose.rs [hunk 2]
```diff
@@ -39,6 +39,7 @@ impl XRJointPose {
         reflect_dom_object(
             Box::new(XRJointPose::new_inherited(&transform, radius)),
             global,
+            CanGc::note(),
```
Comment (simonwuelker) [inline components/script/dom/webxr/xrjointpose.rs:42]: There is already a `can_gc` argument for this function

File: components/script/dom/webxr/xrinputsource.rs [hunk 3]
```diff
@@ -82,6 +82,7 @@ impl XRInputSource {
         let source = reflect_dom_object(
             Box::new(XRInputSource::new_inherited(global, session, info, can_gc)),
             global,
+            CanGc::note(),
```
Comment (simonwuelker) [inline components/script/dom/webxr/xrinputsource.rs:85]: There is already a `can_gc` argument for this function

File: components/script/dom/webgpu/gpudevice.rs [hunk 4]
```diff
@@ -166,6 +166,7 @@ impl GPUDevice {
                 lost_promise,
             )),
             global,
+            CanGc::note(),
```
Comment (simonwuelker) [inline components/script/dom/webgpu/gpudevice.rs:169]: There already is a `can_gc` argumentn to this function

File: components/script/dom/webglrenderingcontext.rs [hunk 5]
```diff
@@ -291,7 +291,7 @@ impl WebGLRenderingContext {
         can_gc: CanGc,
     ) -> Option<DomRoot<WebGLRenderingContext>> {
         match WebGLRenderingContext::new_inherited(window, canvas, webgl_version, size, attrs) {
-            Ok(ctx) => Some(reflect_dom_object(Box::new(ctx), window)),
+            Ok(ctx) => Some(reflect_dom_object(Box::new(ctx), window, CanGc::note())),
```
Comment (simonwuelker) [inline components/script/dom/webglrenderingcontext.rs:294]: there already is a `can_gc` argument for this function

File: components/script/dom/webgl2renderingcontext.rs [hunk 6]
```diff
@@ -192,7 +192,7 @@ impl WebGL2RenderingContext {
         can_gc: CanGc,
     ) -> Option<DomRoot<WebGL2RenderingContext>> {
         WebGL2RenderingContext::new_inherited(window, canvas, size, attrs, can_gc)
-            .map(|ctx| reflect_dom_object(Box::new(ctx), window))
+            .map(|ctx| reflect_dom_object(Box::new(ctx), window, CanGc::note()))
```
Comment (simonwuelker) [inline components/script/dom/webgl2renderingcontext.rs:195]: there already is a `can_gc` argument for this function

File: components/script/dom/bindings/iterable.rs [hunk 7]
```diff
@@ -72,7 +72,7 @@ impl<T: DomObjectIteratorWrap + JSTraceable + Iterable> IterableIterator<T> {
             iterable: Dom::from_ref(iterable),
             index: Cell::new(0),
         });
-        reflect_dom_object(iterator, &*iterable.global())
+        reflect_dom_object(iterator, &*iterable.global(), CanGc::note())
```
[the remark this answers is not in the document]
Comment (simonwuelker) [reply components/script/dom/bindings/iterable.rs:75]: > I think these change will not have any effect

These changes (correctly) indicate that `reflect_dom_object` can cause a garbage collection. In the future we would want to propagate the `CanGc` from all callers, but that is a huge change, and I think it's perfectly reasonable to only update the signature of `reflect_dom_object` for now.

File: components/script/dom/audiolistener.rs [hunk 8]
```diff
@@ -155,7 +156,7 @@ impl AudioListener {
     #[allow(crown::unrooted_must_root)]
     pub fn new(window: &Window, context: &BaseAudioContext) -> DomRoot<AudioListener> {
         let node = AudioListener::new_inherited(window, context);
-        reflect_dom_object(Box::new(node), window)
+        reflect_dom_object(Box::new(node), window, CanGc::note())
```
[the remark this answers is not in the document]
Comment (jdm) [reply components/script/dom/audiolistener.rs:159]: Since this PR is so easy to hit merge conflicts I suggest the TODO.

**Pull request merged**